demos/http3/ossl-nghttp3-demo-server.c

```diff
@@ -335,169 +476,325 @@ static int read_from_ssl_ids(nghttp3_conn *h3conn, struct h3ssl *h3ssl)
         return 0;
     }
 
-    /* We have something */
-    item = items;
-    /* SSL_accept_stream if anyway */
-    if ((item->revents & SSL_POLL_EVENT_ISB) ||
-        (item->revents & SSL_POLL_EVENT_ISU)) {
-        SSL *stream = SSL_accept_stream(ssl_ids[0].s, 0);
-        uint64_t id;
-        int r;
-
-        if (stream == NULL) {
-            return -1; /* something is wrong */
-        }
-        id = SSL_get_stream_id(stream);
-        printf("=> Received connection on %lld %d\n", (unsigned long long) id,
-               SSL_get_stream_type(stream));
-        add_id(id, stream, h3ssl);
-        if (SSL_get_stream_type(stream) == SSL_STREAM_TYPE_BIDI) {
-            /* bidi that is the id  where we have to send the response */
-            printf("=> Received connection on %lld ISBIDI\n",
-                   (unsigned long long) id);
-            h3ssl->id_bidi = id;
-
-            /* XXX use it to restart to end_headers_received */
-            h3ssl->end_headers_received = 0;
-            h3ssl->datadone = 0;
-            h3ssl->close_done = 0;
-            h3ssl->done = 0;
-            h3ssl->restart = 1; /* Checked in wait_close loop */
-        } else {
-           set_id_status(id, CLIENTUNIOPEN, h3ssl);
+    /* reset the states */
+    h3ssl->new_conn = 0;
+    h3ssl->restart = 0;
+    h3ssl->done = 0;
+
+    /* Process all the item we have polled */
+    item = NULL;
+    for (i = 0; i < numitem; i++) {
+        SSL *s;
+
+        if (!item)
+            item = items;
+        else
+            item++;
```

Comment:
Perhaps we can simplify code here to something like this:
```
for (i = 0, item = items; i < numitem; i++, items++) {
```
line 485, and lines 489 - 492 can be deleted.